web/modules/weather_data/src/Service/WeatherAlertTrait.php

```diff
@@ -190,6 +198,259 @@ public function getAlerts($grid, $point, $self = false)
+        }
+
+        // Filter out any periods whose duration was
+        // computed to zero. This can happen when
+        // the diff between a period start and an old
+        // alert end time is only in seconds (ie zero minutes
+        // and zero days)
+        return array_filter($alertPeriods, function ($alertPeriod) {
+            return $alertPeriod["duration"] > 0;
+        });
+    }
+
+    /**
+     * Compute a DateInterval in hours
+     *
+     */
+    private function dateDiffInHours(\DateInterval $diff)
+    {
+        $days = $diff->days * 24;
+        return $diff->h + $days;
+    }
+
+    /**
+     * Determine whether a given DateTime is within two others
+     */
+    private function dateTimeIsWithin($subject, $beginning, $end)
+    {
+        return $beginning <= $subject && $subject < $end;
+    }
+
+    /**
+     * Determine if a subject beginning and end datetime
+     * encompasses (both starts before and ends after)
+     * a comparison beginning and end datetime
+     */
+    private function dateTimeEncompasses(
+        $subjectStart,
+        $subjectEnd,
+        $comparisonStart,
+        $comparisonEnd,
+    ) {
+        return $subjectStart <= $comparisonStart &&
+            $subjectEnd >= $comparisonEnd;
+    }
+
+    /**
+     * Compute the duration of the alert
+     *
+     * The duration is a function of the DateTime
+     * diff, but adjusted for the overall period
+     * range.
+     * Note that DateInterval is "special"
+     * about how it computes diffs
+     */
+    private function calculateAlertDuration($onsetTime, $endTime, $max)
+    {
+        $alertDiff = $endTime->diff($onsetTime, true);
+        $alertDuration = $this->dateDiffInHours($alertDiff);
+
+        // If there are leftover minutes, we add
+        // an hour for coverage purposes
+        if ($alertDiff->m) {
+            $alertDuration += 1;
+        }
+
+        return min($max, $alertDuration);
+    }
+
+    /**
+     * Find relevant alert info in a list of hourly periods
+     *
+     * Given a list of hourly forecast periods and an alert,
+     * attempt to find alert alignment info within any of
+     * those periods.
+     * Return false if nothing is found
+     */
+    private function getAlertInfoInPeriods(
+        $alert,
+        $alertOnset,
+        $alertEnd,
+        $periods,
+        $timezone,
+    ) {
+        foreach ($periods as $periodIndex => $period) {
+            $periodStartTime = self::turnToDate(
+                $period["timestamp"],
+                $timezone,
+            );
+            $periodEndTime = $periodStartTime->modify("+ 1 hour");
+
+            $onsetIsWithinPeriod = $this->dateTimeIsWithin(
+                $alertOnset,
+                $periodStartTime,
+                $periodEndTime,
+            );
+            $endIsWithinPeriod = $this->dateTimeIsWithin(
+                $alertEnd,
+                $periodStartTime,
+                $periodEndTime,
+            );
```

Comment:
Am I misreading, or is this not being used?

Reply:
Good catch! Nixed in https://github.com/weather-gov/weather.gov/pull/847/commits/f7f1d87aebaeef797a275ed431079ca75ca44429